.github/workflows/ui.yml

```diff
@@ -57,3 +57,39 @@ jobs:
         run: |
           . venv/bin/activate
           pnpm run test:ct
+  chromatic-deployment:
+    runs-on: ubuntu-latest
+    if: ${{ !contains( github.event.pull_request.labels.*.name, 'no-visual-update') }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+            fetch-depth: 0
+      - name: install gradio
+        run: python -m pip install gradio==$(cat gradio/version.txt)
+      - name: generate theme.css
+        run: python scripts/generate_theme.py
+      - name: install dependencies
+        uses: "./.github/actions/install-all-deps"
+        with:
+          always-install-pnpm: true
+          skip_build: 'true'
+      - name: build storybook
+        run: pnpm build-storybook --quiet 
+      - name: publish to chromatic
+        id: publish-chromatic
+        uses: chromaui/action@v1
+        with:
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: post deployment link to PR
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `Chromatic build successful :tada:
+              * [Storybook Preview](${{steps.publish-chromatic.outputs.storybookUrl}})
+              * [Build Review](${{steps.publish-chromatic.outputs.buildUrl}})`
+            })
```

Comment:
I think this will make a new comment everytime the action is run. It would be good to comment once and update the existing comment. We are using an action to do this in another workflow. You can see it here: https://github.com/gradio-app/gradio/blob/main/.github/workflows/deploy-pr-to-spaces.yml#L54-L61

All we need to do is choose a different `comment_tag`, change the message, and pass in the  github token and i think it should work.